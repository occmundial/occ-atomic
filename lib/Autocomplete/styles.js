import idx from 'idx';
import { createUseStyles } from 'react-jss';

import spacing from '../subatomic/spacing';

export default createUseStyles({
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: props => idx(props, _ => _.textfieldProps.label) || idx(props, _ => _.textfieldProps.lockHeight) ? spacing.xLarge : 40,
        left: 0,
        width: '100%'
    }
});
