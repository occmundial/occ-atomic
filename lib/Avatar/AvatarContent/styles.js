import fonts from '../../tokens/fonts.json';
import colors from '../../tokens/colors.json';
import borderRadius from '../../tokens/borderRadius.json';
import legacyFont from '../../subatomic/fonts';

const { avatar, text, border } = colors;
const transition = 'all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s';

const font = fonts['body-regular-strong'];

export default {
	wrap: {
		position: 'relative',
		width: '100%',
		height: '100%'
	},
	cont: {
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		borderRadius: borderRadius['br-full'],
		overflow: 'hidden',
		background: avatar.bg.default,
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center'
	},
	capital: {
		display: 'inline-block',
		font: `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${legacyFont.body}`,
		color: text.indigo.primary,
		gridRowStart: '1',
		gridColumnStart: '1',
		textAlign: 'center'
	},
	disabled: {
		pointerEvents: 'none',
		background: avatar.bg.disabled,
		'& $camera': {
			opacity: 1
		}
	},
	disabledCapital: {
		color: text.corp.disabled
	  },
	clickable: {
		transition,
			'&:hover': {
				background: avatar.bg.hover
			},
			'&:active': {
				background: avatar.bg.active
			}
	  },
	editable: {
		'&:hover $camera': {
			opacity: 1
		},
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	},
	person: {
		gridRowStart: 1,
		gridColumnStart: 1,
		transition
	},
	  photo: {
		border: `1px solid ${border.default.subtle}`
	},
	hoverPhoto: {
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	  },
	camera: {
		gridRowStart: 1,
		gridColumnStart: 1,
		opacity: 0,
		transition
	},
	overlay: {
		borderRadius: '50%',
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center',
		transition,
		border: `1px solid transparent`
	  }
};
