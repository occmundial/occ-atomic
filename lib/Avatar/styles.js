import { createUseStyles } from 'react-jss';

export default createUseStyles({
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
});
