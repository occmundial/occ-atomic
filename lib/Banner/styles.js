import spacing from '../subatomic/spacing';
import colors from '../subatomic/colors';
import grid from '../subatomic/grid';

export default {
  container: { padding: spacing.small, backgroundColor: colors.ink },
  textBanner: { display: 'inline-block' },
  textFlex: {
    textAlign: 'center',
    [`@media screen and (max-width:${grid.xs}px)`]: { textAlign: 'start' },
  },
  iconFlex: { [`@media screen and (max-width:${grid.xs}px)`]: { padding: spacing.small } },
  icon: { cursor: 'pointer' },
};
