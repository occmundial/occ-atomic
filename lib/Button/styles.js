import legacyColors from '../subatomic/colors';
import legacyFont from '../subatomic/fonts';
import icons from '../subatomic/icons';

import colors from '../tokens/colors.json';
import spacing from '../tokens/spacing.json';
import borderRadius from '../tokens/borderRadius.json';
import fonts from '../tokens/fonts.json';
import shadows from '../tokens/shadows.json';
import { legacyIcons } from '../subatomic/mappedIcons';

const { sec, secDark, grey500, secLight } = legacyColors;

const sm = fonts['button-small'];
const md = fonts['button-medium'];
const lg = fonts['button-large'];

const renderIcon = (props, color) => {
    if (!legacyIcons[props.icon]) return '';
    if (props.icon) {
        return icons.base(icons[props.icon].icon(color));
    } else if (props.iconRight) {
        return icons.base(icons[props.iconRight].icon(color));
    }
    return '';
};

export default {
    btn: {
        display: 'inline-block',
        boxSizing: 'border-box',
        position: 'relative',
        maxWidth: '100%',
        marginBottom: 0,
        border: 0,
        padding: [spacing['size-3'], spacing['size-4']],
        borderRadius: borderRadius['br-xs'],
        font: `${sm.fontWeight} ${sm.fontSize}/${sm.lineHeight} ${legacyFont.body}`,
        textAlign: 'center',
        textDecoration: 'none',
        whiteSpace: 'nowrap',
        transition: '0.3s all',
        cursor: 'pointer',
        userSelect: 'none',
        touchAction: 'manipulation',
        '&:focus': {
            outline: 'none'
        },
        '&:hover': {
            textDecoration: 'none'
        }
    },
    cont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        pointerEvents: 'none'
    },
    loadCont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'absolute',
        top: 0,
        left: 0,
        pointerEvents: 'none'
    },
    // Themes
    primary: {
        background: colors.button.primary.bg.default,
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.primary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.primary.bg.hover
        },
        '&:active': {
            background: colors.button.primary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.primary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-pink']
        },
        '&$disabled': {
            background: colors.button.primary.bg.disabled,
            color: colors.text.white.secondary,
            cursor: 'not-allowed'
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    secondary: {
        background: colors.button.secondary.bg.default,
        color: colors.text.indigo.primary,
        '&:hover': {
            background: colors.button.secondary.bg.hover
        },
        '&:active': {
            background: colors.button.secondary.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '&$disabled': {
            background: colors.button.secondary.bg.disabled,
            color: colors.text.indigo.secondary
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiary: {
        background: 'transparent',
        color: colors.text.indigo.primary,
        outline: `2px solid ${colors.button.tertiary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.indigo.secondary,
            outline: `2px solid ${colors.button.tertiary.border.disabled}`
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiaryWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.tertiary.border.inverse.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.inverse.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.inverse.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    ghostPink: {
        background: 'transparent',
        color: sec,
        border: 'none',
        paddingLeft: [0, '!important'],
        paddingRight: [0, '!important'],
        '&:hover, &:active': {
            color: secDark,
            '& span span': {
                background: props => renderIcon(props, [secDark])
            },
        },
        '& span span': {
            background: props => renderIcon(props, [sec])
        },
        '&$disabled': {
            background: `transparent !important`,
            color: `${secLight} !important`,
            border: `none !important`,
            '& span span': {
                background: props => renderIcon(props, [grey500])
            }
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([sec]))
        }
    },
    ghostGrey: {
        background: 'transparent',
        color: colors.text.corp.secondary,
        '&:hover': {
            background: colors.button.ghost.bg.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-corp']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.default.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.corp.disabled
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.default.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.default.default]))
        }
    },
    ghostWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        '&:hover': {
            background: colors.button.ghost.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.inverse.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    loading: {
        pointerEvents: 'none',
        '& $cont': {
            visibility: 'hidden'
        }
    },
    disabled: {
        cursor: 'default',
        pointerEvents: 'none',
    },
    // Sizes
    md: {
        font: `${md.fontWeight} ${md.fontSize}/${md.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-5']],
        '&$iconOnly': {
            padding: [spacing['size-2'], spacing['size-2']]
        }
    },
    lg: {
        font: `${lg.fontWeight} ${lg.fontSize}/${lg.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-6']],
        '&$iconOnly': {
            padding: [spacing['size-3'], spacing['size-3']]
        }
    },
    // Block
    block: {
        display: 'block',
        width: '100%'
    },
    // Icon
    icon: {
        marginRight: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconRight: {
        width: 16,
        height: 16,
        marginLeft: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconOnly: {
        padding: [spacing['size-2'], spacing['size-2']],
        '& span span, & svg': {
            marginRight: 0
        }
    },
    loadIcon: {
        width: 24,
        height: 24,
        fill: colors.icon.inverse.default
    },
    round: {
        borderRadius: '50%'
    }
};
