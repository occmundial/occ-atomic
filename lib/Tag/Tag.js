import React from 'react';
import PropTypes from 'prop-types';
import Flexbox from '../Flexbox';
const tagSizes = {
	standard: 'standard',
	medium: 'medium',
	big: 'big'
};
/** Tag component with different themes. The tag cannot trigger some actions on click. */
const Tag = ({
	classes,
	children,
	className,
	style,
	theme,
	iconName,
	size
}) => (
	<label
		className={`${classes.tag} ${size ? classes[size] : classes.standard} ${
			theme ? classes[theme] : classes.default
		} ${className || ''}`}
		style={style}
	>
		 <Flexbox display="flex" alignItems="center" wrap="noWrap">
			{iconName && (
				<span
					className={`${classes.icon} ${
						size ? classes[`${size}Icon`] : classes.standardIcon
					} ${theme ? classes[`${theme}Icon`] : classes.defaultIcon}`}
				/>
			)}
			<span
				className={`${classes.tagText} ${
					size ? classes[`${size}TagText`] : classes.standardTagText
				}`}
			>
				{children}
			</span>
		 </Flexbox>
	</label>
);

Tag.defaultProps = {
	theme: 'default',
	size: tagSizes.standard
};

Tag.propTypes = {
	classes: PropTypes.object,
	children: PropTypes.node,
	/** Select a theme for the Tag. */
	theme: PropTypes.string,
	/** Select an icon for the Tag. */
	iconName: PropTypes.string,
	className: PropTypes.string,
	style: PropTypes.object,
	/** Size in Tag. */
	size: PropTypes.oneOf([tagSizes.standard, tagSizes.medium, tagSizes.big])
};

export default Tag;
