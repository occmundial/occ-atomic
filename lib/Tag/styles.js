import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import icons from '../subatomic/icons';
import hexToRgba from 'hex-rgba';
import borderRadius from '../tokens/borderRadius.json';
import spacing from '../tokens/spacing.json';
import fontsNew from '../tokens/fonts.json';
import colorsNew from '../tokens/colors.json';

const { tag, text, icon } = colorsNew;

export const objectToFontValue = (font) =>
    `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${fonts.body}`;

const { grey900, grey100, textLink, bgWhite } = colors;

export default {
	tag: {
		boxSizing: 'border-box',
		lineHeight: '12px',
		textAlign: 'center',
		borderRadius: borderRadius['br-full'],
		position: 'relative',
		overflow: 'hidden',
		display: 'inline-block'
	},
	standard: {
		padding: [spacing['size-0'], spacing['size-2']],
	},
	medium: {
		padding: [spacing['size-2'], spacing['size-3']],
	},
	big: {
		minHeight: spacing['size-6'],
		padding: [spacing['size-2'], spacing['size-4']],
	},
	tagText: {
		boxSizing: 'border-box',
		font: objectToFontValue(fontsNew['heading-tag']),
		fontStyle: 'normal',
		fontStretch: 'normal',
		letterSpacing: 'normal',
		textAlign: 'center',
		position: 'relative',
		overflow: 'hidden',
		display: 'inline-block',
		paddingRight: props => (props.iconName ? spacing['size-1'] : 0)
	},
	default: {
		color: grey900,
		background: grey100
	},
	basic: {
		color: grey900,
		background: bgWhite
	},
	link: {
		color: textLink,
		background: hexToRgba(textLink, 10)
	},

	featured: {
		color: text.white.primary,
		background: tag.featured.bg,
		border: `1px solid ${tag.featured.border}`,
	},
	promo: {
		color: text.white.primary,
		background: tag.promo.bg,
		border: `1px solid ${tag.promo.border}`,
	},
	success: {
		color: text.success,
		background: tag.success.bg,
		border: `1px solid ${tag.success.border}`,
	  },
	  info: {
		color: text.indigo.primary,
		background: tag.info.bg,
		border: `1px solid ${tag.info.border}`,
	  },
	  warning: {
		color: text.warning,
		background: tag.warning.bg,
		border: `1px solid ${tag.warning.border}`,
	  },
	  error: {
		color: text.error,
		background: tag.error.bg,
		border: `1px solid ${tag.error.border}`,
	  },
	/* Icon with themes */
	icon: {
		display: 'inline-block',
		marginRight: spacing['size-1'],
		content: '""',
		paddingBottom: 0,
		left: 0,
		backgroundRepeat: 'no-repeat'
	},
	standardIcon: {
		width: spacing['size-3'],
		height: spacing['size-3']
	},
	bigIcon: {
		width: spacing['size-4'],
		height: spacing['size-4']
	},
	defaultIcon: {
		background: props =>
			props.iconName ? icons.base(icons[props.iconName].icon([grey900])) : ''
	},
	basicIcon: {
		background: props =>
			props.iconName ? icons.base(icons[props.iconName].icon([grey900])) : ''
	},
	linkIcon: {
		background: props =>
			props.iconName ? icons.base(icons[props.iconName].icon([textLink])) : ''
	},
	featuredIcon: {
		background: props =>
			props.iconName ? icons.base(icons[props.iconName].icon([icon.inverse.default])) : ''
	},
	promoIcon: {
		background: props =>
			props.iconName ? icons.base(icons[props.iconName].icon([icon.inverse.default])) : ''
	},

	successIcon: {
		background: props =>
			props.iconName ? icons.base(icons[props.iconName].icon([icon.success])) : ''
	},
	infoIcon: {
		background: props =>
			props.iconName ? icons.base(icons[props.iconName].icon([icon.brand.bold])) : ''
	},
	warningIcon: {
		background: props =>
			props.iconName
				? icons.base(icons[props.iconName].icon([icon.warning]))
				: ''
	},
	errorIcon: {
		background: props =>
			props.iconName ? icons.base(icons[props.iconName].icon([icon.error])) : ''
	}
};
