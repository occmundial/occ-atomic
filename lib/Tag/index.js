import injectSheet from 'react-jss';
import Tag from './Tag';
import styles from './styles';

export default injectSheet(styles)(Tag);
