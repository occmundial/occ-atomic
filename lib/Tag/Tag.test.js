import React from 'react';
import { shallow } from 'enzyme';
import Tag from './Tag';
import styles from './styles';
import TagJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe('Tag', () => {
	it('matches the snapshot', () => {
		const wrapper = shallow(<Tag classes={classes}>Tag</Tag>);
		expect(wrapper).toMatchSnapshot();
	});

	it('shows the children content', () => {
		const wrapper = shallow(<Tag classes={classes}>Tag</Tag>);
		expect(wrapper.text()).toEqual('Tag');
	});

	it('has the right theme', () => {
		const wrapper = shallow(
			<Tag classes={classes} theme="default">
				Tag
			</Tag>
		);
		expect(wrapper.prop('className')).toEqual('tag default ');
	});

	it('render the span when necessary', () => {
		const wrapper = shallow(<Tag classes={classes}>Tag Test</Tag>);
		expect(wrapper.find('span').length).toBe(1);
		wrapper.setProps({ iconName: 'check' });
		expect(wrapper.find('span').length).toBe(2);
	});

	it('shows an icon', () => {
		const wrapper = shallow(<Tag classes={classes}>Tag Test</Tag>);
		expect(wrapper.find('.icon').length).toBe(0);
		wrapper.setProps({ iconName: 'check' });
		expect(wrapper.find('.icon').length).toBe(1);
	});
});

describe('TagJSS', () => {
	it('matches the snapshot', () => {
		const wrapper = shallow(<TagJSS>Tag</TagJSS>);
		expect(wrapper).toMatchSnapshot();
	});
});

describe('Tag styles', () => {
	it('matches the snapshot', () => {
		expect(styles).toMatchSnapshot();
	});

	it('returns the right icons', () => {
		let props = { iconName: 'testing' };
		expect(styles.defaultIcon.background(props)).toBe(
			'url(data:image/svg+xml;base64,dGVzdGluZyAtICMyMjIyMjI=)'
		);
	});
});
