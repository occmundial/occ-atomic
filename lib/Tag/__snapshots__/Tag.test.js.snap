// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tag
    classes={
      Object {
        "default": "default",
        "defaultIcon": "defaultIcon",
        "error": "error",
        "errorIcon": "errorIcon",
        "icon": "icon",
        "info": "info",
        "infoIcon": "infoIcon",
        "link": "link",
        "linkIcon": "linkIcon",
        "success": "success",
        "successIcon": "successIcon",
        "tag": "tag",
        "tagText": "tagText",
        "warning": "warning",
        "warningIcon": "warningIcon",
      }
    }
    theme="default"
  >
    Tag
  </Tag>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <span
          className="tagText"
        >
          Tag
        </span>,
      ],
      "className": "tag default ",
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Tag",
          "className": "tagText",
        },
        "ref": null,
        "rendered": "Tag",
        "type": "span",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <span
            className="tagText"
          >
            Tag
          </span>,
        ],
        "className": "tag default ",
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tag",
            "className": "tagText",
          },
          "ref": null,
          "rendered": "Tag",
          "type": "span",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Tag styles matches the snapshot 1`] = `
Object {
  "default": Object {
    "background": "#f5f5f5",
    "color": "#222222",
  },
  "defaultIcon": Object {
    "background": [Function],
  },
  "error": Object {
    "background": "rgba(219, 55, 55, 0.1)",
    "color": "#db3737",
  },
  "errorIcon": Object {
    "background": [Function],
  },
  "icon": Object {
    "backgroundRepeat": "no-repeat",
    "content": "\\"\\"",
    "display": "inline-block",
    "height": 13,
    "left": 0,
    "marginRight": 4,
    "paddingBottom": 0,
    "position": "relative",
    "top": 0,
    "width": 12,
  },
  "info": Object {
    "backgroundColor": "rgba(87, 54, 171, 0.1)",
    "color": "#5736ab",
  },
  "infoIcon": Object {
    "background": [Function],
  },
  "link": Object {
    "background": "rgba(9, 70, 203, 0.1)",
    "color": "#0946CB",
  },
  "linkIcon": Object {
    "background": [Function],
  },
  "success": Object {
    "background": "rgba(0, 149, 55, 0.1)",
    "color": "#009537",
  },
  "successIcon": Object {
    "background": [Function],
  },
  "tag": Object {
    "borderRadius": 16,
    "boxSizing": "border-box",
    "display": "inline-block",
    "height": 16,
    "lineHeight": "12px",
    "overflow": "hidden",
    "paddingBottom": 2,
    "paddingLeft": 8,
    "paddingRight": 8,
    "paddingTop": 1,
    "position": "relative",
    "textAlign": "center",
  },
  "tagText": Object {
    "boxSizing": "border-box",
    "display": "inline-block",
    "fontFamily": "'hk_grotesk', sans-serif",
    "fontSize": 10,
    "fontStretch": "normal",
    "fontStyle": "normal",
    "fontWeight": "600",
    "letterSpacing": "normal",
    "lineHeight": "13px",
    "overflow": "hidden",
    "paddingRight": [Function],
    "position": "relative",
    "textAlign": "center",
  },
  "warning": Object {
    "background": "rgba(255, 197, 73, 0.1)",
    "color": "#ffc549",
  },
  "warningIcon": Object {
    "background": [Function],
  },
}
`;

exports[`TagJSS matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Tag)
    theme="default"
  >
    Tag
  </Jss(Tag)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Tag",
      "classes": Object {
        "default": "Tag-default-0-1-24 Tag-default-0-1-3",
        "defaultIcon": "Tag-defaultIcon-0-1-17 Tag-defaultIcon-0-1-10",
        "error": "Tag-error-0-1-28 Tag-error-0-1-7",
        "errorIcon": "Tag-errorIcon-0-1-21 Tag-errorIcon-0-1-14",
        "icon": "Tag-icon-0-1-30 Tag-icon-0-1-9",
        "info": "Tag-info-0-1-25 Tag-info-0-1-4",
        "infoIcon": "Tag-infoIcon-0-1-18 Tag-infoIcon-0-1-11",
        "link": "Tag-link-0-1-29 Tag-link-0-1-8",
        "linkIcon": "Tag-linkIcon-0-1-22 Tag-linkIcon-0-1-15",
        "success": "Tag-success-0-1-26 Tag-success-0-1-5",
        "successIcon": "Tag-successIcon-0-1-19 Tag-successIcon-0-1-12",
        "tag": "Tag-tag-0-1-23 Tag-tag-0-1-1",
        "tagText": "Tag-tagText-0-1-16 Tag-tagText-0-1-2",
        "warning": "Tag-warning-0-1-27 Tag-warning-0-1-6",
        "warningIcon": "Tag-warningIcon-0-1-20 Tag-warningIcon-0-1-13",
      },
      "theme": "default",
    },
    "ref": null,
    "rendered": "Tag",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Tag",
        "classes": Object {
          "default": "Tag-default-0-1-24 Tag-default-0-1-3",
          "defaultIcon": "Tag-defaultIcon-0-1-17 Tag-defaultIcon-0-1-10",
          "error": "Tag-error-0-1-28 Tag-error-0-1-7",
          "errorIcon": "Tag-errorIcon-0-1-21 Tag-errorIcon-0-1-14",
          "icon": "Tag-icon-0-1-30 Tag-icon-0-1-9",
          "info": "Tag-info-0-1-25 Tag-info-0-1-4",
          "infoIcon": "Tag-infoIcon-0-1-18 Tag-infoIcon-0-1-11",
          "link": "Tag-link-0-1-29 Tag-link-0-1-8",
          "linkIcon": "Tag-linkIcon-0-1-22 Tag-linkIcon-0-1-15",
          "success": "Tag-success-0-1-26 Tag-success-0-1-5",
          "successIcon": "Tag-successIcon-0-1-19 Tag-successIcon-0-1-12",
          "tag": "Tag-tag-0-1-23 Tag-tag-0-1-1",
          "tagText": "Tag-tagText-0-1-16 Tag-tagText-0-1-2",
          "warning": "Tag-warning-0-1-27 Tag-warning-0-1-6",
          "warningIcon": "Tag-warningIcon-0-1-20 Tag-warningIcon-0-1-13",
        },
        "theme": "default",
      },
      "ref": null,
      "rendered": "Tag",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
